### PR TITLE
MGMT-2695 Move logs upload command to separate cmd

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: internal/host/logscmd.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		baseCmd:           baseCmd{log: log},
+		instructionConfig: instructionConfig,
+	}
+}
+
+func (i *logsCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models.Step, error) {
+	// added to run upload logs if install command fails
+	if host.LogsCollectedAt != strfmt.DateTime(time.Time{}) {
+		return nil, nil
+	}
```
Comment (ybettan) [inline internal/host/logscmd.go:31]: It is possible that hosts logs were already sent to service but installer didn't finish yet?
In this case you may have host logs already but installer failed, in this case you won't get the installer-gather logs ?!

File: internal/host/logscmd.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	logsCommand, err := CreateUploadLogsCmd(host, i.instructionConfig.ServiceBaseURL,
+		i.instructionConfig.InventoryImage, i.instructionConfig.SkipCertVerification,
+		false, true)
```
Comment (ybettan) [inline internal/host/logscmd.go:34]: Don't we want to parameterise the `withInstallerGatherLogs` here? You assume we always run installer-gather (which is true in assisted-service) but if we already made it a separate command maybe we want to make it flexible no?!

**Comment by romfreiman** [comment]
@tsorya @ybettan just to verify we dont break competability here - if there is a discovery iso that is running, and we upgrade, what will happen? (even before we triggered the installation)

**Pull request merged**